tests/tests/serde.rs

```diff
@@ -148,14 +157,210 @@ mod tests {
                 ];
                 tokens.extend(point_compressed.iter().copied().map(Token::U8));
                 tokens.push(Token::SeqEnd);
-                serde_test::assert_de_tokens(&Compressed(point).readable(), &tokens);
+                serde_test::assert_de_tokens(&Compact(point).readable(), &tokens);
+                serde_test::assert_de_tokens(&PreferCompact(point).readable(), &tokens);
+            }
+
+            // Seq-encoded struct, human-readable (hex-encoded)
+            serde_test::assert_de_tokens(
+                &point.readable(),
+                &[
+                    Token::Seq { len: Some(2) },
+                    Token::Str(E::CURVE_NAME),
+                    Token::Str(point_uncompressed_hex),
+                    Token::SeqEnd,
+                ],
+            );
+            // Seq-encoded struct, binary (bytes-encoded)
+            serde_test::assert_de_tokens(
+                &point.readable(),
+                &[
+                    Token::Seq { len: Some(2) },
+                    Token::Str(E::CURVE_NAME),
+                    Token::Bytes(point_uncompressed),
+                    Token::SeqEnd,
+                ],
+            );
+            // Seq-encoded struct, binary (seq-encoded)
+            {
+                let mut tokens = vec![
+                    Token::Seq { len: Some(2) },
+                    Token::Str(E::CURVE_NAME),
+                    Token::Seq { len: None },
+                ];
+                tokens.extend(point_uncompressed.iter().copied().map(Token::U8));
+                tokens.extend([Token::SeqEnd, Token::SeqEnd]);
+
+                serde_test::assert_de_tokens(&point.readable(), &tokens);
+            }
+
+            // PreferCompact doesn't support seq-encoded structs
+            serde_test::assert_de_tokens_error::<serde_test::Compact<PreferCompact<Point<E>>>>(
```

Comment:
I don't know how serde tests work, but why is PreferCompact embedded in Compact here?

Reply:
serde_test::Comact says that data should be deserialized with `human_readable = false` (as opposite to `serde_test::Readable`)

Reply:
if deserialization logic relies on `is_human_readable` flag, but the type isn't wrapped into either `serde_test::{Compact, Readable}`, assertion panics